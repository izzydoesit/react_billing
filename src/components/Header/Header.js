// @flow 

import React from 'react';
import './Header.css';

const Header = () =>

  <div className="header-wrapper">
    <header>
      <h1 className="page-title">Welcome to the Clark Take-Home Code Challenge</h1>
    </header>
  </div>

export default Header;
