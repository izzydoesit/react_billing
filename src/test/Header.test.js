// @flow

import React from 'react';
import Header from '../components/Header/Header';
import {shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<Header />', () => {
  let header;

  beforeEach(() => {
    header = shallow(<Header />);
  });

  it('renders correctly', () => {
    expect(shallowToJson(header)).toMatchSnapshot();
  });

  it('renders a <div> element', () => {
    expect(header.find('div.header-wrapper').length).toEqual(1);
  });

  it('renders an <h1> element', () => {
    expect(header.find('h1').length).toEqual(1);
    expect(header.find('h1').text()).toEqual('Welcome to the Clark Take-Home Code Challenge');
  });
});
