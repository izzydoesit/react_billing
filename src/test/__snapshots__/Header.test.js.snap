// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<div
  className="header-wrapper"
>
  <header>
    <h1
      className="page-title"
    >
      Welcome to the Clark Take-Home Code Challenge
    </h1>
  </header>
</div>
`;
